src/java/org/apache/cassandra/service/StorageService.java

```diff
@@ -1113,6 +1121,18 @@ public void doAuthSetup(boolean async)
         }
     }
 
+    public void doAutoRepairSetup()
+    {
+        AutoRepairService.setup();
+        if (DatabaseDescriptor.getAutoRepairConfig().isAutoRepairSchedulingEnabled())
+        {
+            logger.info("Enabling auto-repair scheduling");
+            AutoRepair.instance.setup();
+        }
+        logger.info("AutoRepair setup complete!");
```

Comment:
nit: should move this into the `if` statement above as if you have the feature disabled it makes it appear as if it was enabled.

Reply:
Incorporated